src/client/datascience/errorHandler/errorHandler.ts

```diff
@@ -20,6 +21,10 @@ export class DataScienceErrorHandler implements IDataScienceErrorHandler {
     ) {}
 
     public async handleError(err: Error): Promise<void> {
+        // Unwrap the errors.
+        if (err instanceof WrappedError && err.originalException && err.originalException instanceof BaseError) {
```

Comment:
This was required, as some errors get wrapped using `WrappedError` instead of re-throwing the same error. Rather than fixing it, i figured I might as well unwrap here to ensure we never fall into this same issue again.